### PR TITLE
Add GitHub Actions

### LINKED ISSUE #50: Using GitHub features to manage repository easily
Github offers a lot of features via special `.github` folder. Do you have any plan to adapt any of them? For example:

- **Pull Request Template**
- **Issue Templates** (multiple)
- **Workflows** (instead of travis)
- **Dependabot** (_beta_ - Check go.mod file and automatically create a pull request if one of the dependency is outdated.)

All of the above used by my repositories. You can check [https://github.com/erdaltsksn/gh-label/tree/master/.github](https://github.com/erdaltsksn/gh-label/tree/master/.github) to see them.
NOTE: This app `gh-label` uses `go-keyring` so I link it.

If you are willing to adapt, I'm ready to work on it and create a pull request for it.

### PR DESCRIPTION
**Pull request by erdaltsksn**
This closes #50

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (mikkeloscar) [inline .github/workflows/go.yml:7]: This should also run on Pull requests?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        os: [ubuntu-16.04, macOS-latest, windows-latest]
+
+    steps:
+    - name: Check out code into $GITHUB_WORKSPACE directory
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.13
```
Comment (mikkeloscar) [inline .github/workflows/go.yml:24]: Let's bump this to latest go `1.14`

File: .github/workflows/go.yml (hunk 1 above)
Comment (erdaltsksn) [reply .github/workflows/go.yml:7]: with `pull_request` , It will run

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+
+    - name: Install Dependencies (Linux)
+      run: |
+        sudo apt-get install python-gnomekeyring
+        sudo apt-get install gnome-keyring
+        dbus-launch /usr/bin/python -c "import gnomekeyring;gnomekeyring.create_sync('login', '');"
+      if: matrix.os == 'ubuntu-16.04'
```
Comment (mikkeloscar) [inline .github/workflows/go.yml:36]: Maybe add a comment with the link to why we must use 16.04 for now.

File: .github/workflows/go.yml (hunk 2 above)
Comment (erdaltsksn) [reply .github/workflows/go.yml]: Done

File: .github/workflows/go.yml (hunk 3 above)
Comment (erdaltsksn) [reply .github/workflows/go.yml:36]: done

File: .github/workflows/go.yml (hunk 1 above)
Comment (mikkeloscar) [reply .github/workflows/go.yml:7]: Don't you just need this: https://github.com/zalando-incubator/kube-ingress-aws-controller/blob/3bf44bedcfefa4d7a220dffeaa5e749933280c4d/.github/workflows/ci.yaml#L2

With `branches: [master]` I think it's limited to only run when merged to master? I don't see that it's running for this PR for example.

Comment (erdaltsksn) [reply .github/workflows/go.yml:7]: `branches: [ master ]` runs on when a pull request created against the `zalando:master `. Creator's branch is not important. There aren't any branch like `dev` that is used before merging to the master.

> I don't see that it's running for this PR for example.

This action will run for next pull requests after it is merged.

Example: https://github.com/erdaltsksn/gh-label/pull/2/checks

I can remove them if you want more permissive configurations.

Comment (mikkeloscar) [reply .github/workflows/go.yml:7]: Thanks for clarifying. That sounds ok for me as-is. Can you sign-off your commits then we're good to go :)

Comment (erdaltsksn) [reply .github/workflows/go.yml:7]: ✅

**Comment by mikkeloscar** [comment]
:+1:

**Comment by erdaltsksn** [comment]
One more code reviewer?

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
Thanks @erdaltsksn , I missed the former notification.

**Pull request merged**